utils/nhs_number_generator.py

```diff
@@ -0,0 +1,103 @@
+"""Generate 10-digit NHS numbers
+"""
+
+from __future__ import unicode_literals
+
+from random import choice, randint
+
+check_digit_weights = {0: 10, 1: 9, 2: 8, 3: 7, 4: 6, 5: 5, 6: 4, 7: 3, 8: 2}
+
+
+def calculate_check_digit(nhs_number):
+    """Given the first 9 or 10 digits of a 10-digit NHS number, calculate the check digit.
+
+    Returns:
+        int: The check digit.
+             Note that this function may return 10, in which case the NHS number is invalid.
+
+    """
+
+    # The procedure for calculating the check digit, according to:
+    # https://www.datadictionary.nhs.uk/data_dictionary/attributes/n/nhs/nhs_number_de.asp
+
+    # Step 1) Multiply each of the first nine digits by a weighting factor
+    products = [int(nhs_number[i]) * check_digit_weights[i] for i in range(9)]
+
+    # Step 2) Add the results of each multiplication together.
+    sum_products = sum(products)
+
+    # Step 3) Divide the total by 11 and establish the remainder.
+    remainder = sum_products % 11
+
+    # Step 4) Subtract the remainder from 11 to give the check digit.
+    eleven_minus_remainder = 11 - remainder
+
+    # If the result is 11 then a check digit of 0 is used.
+    if eleven_minus_remainder == 11:
+        return 0
+
+    # If the result is 10 then the NHS number is invalid.
+    return eleven_minus_remainder
+
+
+def random_nhs_number_generator(
```

Comment:
These ranges don't need to be default arguments. Can move up-top as a constant, with check_digit_weights.